runtime/syntax/python.vim

```diff
@@ -159,9 +159,9 @@ syn match   pythonEscape	"\\$"
 if !exists("python_no_number_highlight")
   " numbers (including longs and complex)
   syn match   pythonNumber	"\<0[oO]\=\o\+[Ll]\=\>"
-  syn match   pythonNumber	"\<0[xX]\x\+[Ll]\=\>"
+  syn match   pythonNumber	"\<0[xX][0-9A-F_]\+[Ll]\=\>"
```

Comment:
This accepts the following _invalid_ literals
```
0xF_
0x__F
0xFF__FF
```
and highlights them as valid hexadecimal integers.
OTOH, it does not highlight _valid_ hexadecimal integers
```
0xf
0xab123
0x_f
```
The expression needs to be a little more precise to avoid leading, trailing, and duplicate underscores and accept the lowercase a-f for hexadecimal letters.
My upcoming change will take care of that.